signer/src/bitcoin/validation.rs

```diff
@@ -2113,118 +2113,119 @@ mod tests {
         };
     }
 
-    #[test_case(
-        vec![1000, 2000, 3000],
-        RollingWithdrawalLimits {
+    /// A helper struct for testing how the code handles withdrawals with
+    /// specific limits.
+    struct WithdrawalLimitsTestCase {
+        /// The withdrawal amounts that are being considered.
+        withdrawal_amounts: Vec<u64>,
+        /// The rolling withdrawal limits to test.
+        rolling_limits: RollingWithdrawalLimits,
+        /// The total withdrawn so far.
+        withdrawn_total: Amount,
+        /// The expected outcome after running validation on the withdrawal
+        /// requests.
+        expected: Result<(), Error>,
```

Comment:
nit: apparently we can use a `test_case` notation for this, eg
https://github.com/stacks-network/sbtc/blob/3308ed0776b3c31faf5a171c0bb188024068495a/sbtc/src/leb128.rs#L376-L385

Reply:
Yeah, just saw that today or yesterday. I think it'll update it but in the other branch.

Reply:
I tried updating it in the other branch and it's not so simple, the return type needs to implement `PartialEq`, which our `Error` type does not do. So :(